server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java

```diff
@@ -71,27 +68,16 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         ClusterRerouteRequest clusterRerouteRequest = createRequest(request);
-
+        settingsFilter.addFilterSettingParams(request);
+        if (clusterRerouteRequest.explain()) {
+            request.params().put("explain", Boolean.TRUE.toString());
+        }
         // by default, return everything but metadata
         final String metric = request.param("metric");
         if (metric == null) {
             request.params().put("metric", DEFAULT_METRICS);
         }
-
-        return channel ->
-            client.admin().cluster().reroute(clusterRerouteRequest, new AcknowledgedRestListener<ClusterRerouteResponse>(channel) {
-                @Override
-                protected void addCustomFields(XContentBuilder builder, ClusterRerouteResponse response) throws IOException {
-                    builder.startObject("state");
-                    settingsFilter.addFilterSettingParams(request);
-                    response.getState().toXContent(builder, request);
```

Comment:
I see that applying the settingsFilter moved up in this method, but how do we know that the request (as Params) is what is used then rendering the clusterstate now in the response? I was following the code but couldn't connect the dots yet, maybe you know?

Reply:
good one, I think it works as I don't think I changed anything substantial, but I meant to add a test for this and I forgot, thanks for reminding me ;)

Reply:
I think that this is what you're looking for: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/RestToXContentListener.java#L47 . The channel.request gets passed in as params to the `toXContent` call

Reply:
Thanks for checking, yes, that looks like how it works. Adding a test wouldn't hurt, but might not me really necessary either. I'm fine with this explanation.